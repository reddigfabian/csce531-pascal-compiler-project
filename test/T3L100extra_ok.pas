(* Test file for 100% level of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L100_ok;

var
   i,j  : Integer;
   f    : Single;
   c    : Char;
   iaa  : array[0..13, 0..11] of Integer;
   raaa : array[2..4, 0..5, 1..4] of Real;
   r    : Real;


procedure print_nl; External;
procedure print_dot; External;
procedure print_even; External;
procedure print_odd; External;
procedure print_j; External;
procedure print_i_j_values; External;
procedure print_oops; External;


begin
   for i := 1 to 10 do
      print_dot;
   print_nl;

   for i := 1 to 100 do begin
      print_dot;
      if i = 10 then break
   end;
   print_nl;

   i := 15;
   for i := 1 to i do
      print_dot;
   print_nl;

   for c := chr(1) to chr(10) do
      print_dot;
   print_nl;

   for i := 0 to 9 do begin
      case i of
	0,2,4,6,8: print_even;
	1,3,5,7,9: print_odd
      end;
      if i = 1 then i := 3;
      for j := 1 to i-1 do
	 if j = 7 then break
	 else print_j;
      if i > 99 then break
   end;
   print_nl;
   print_i_j_values;

   for j := 10 downto 1 do
      print_i_j_values;

   for i := 0 to 9 do;
   print_oops;

   for i := 10 downto 0 do begin
      case i of
	0,2,4,6,8,10: print_even;
	1,3,7,9     : print_odd;
	5           : break;
      end; { case }
      print_oops
   end

end.
