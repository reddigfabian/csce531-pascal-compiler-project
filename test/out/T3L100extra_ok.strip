b_global_decl (J, alignment = 4, size = 4)
b_global_decl (I, alignment = 4, size = 4)
b_global_decl (F, alignment = 4, size = 4)
b_global_decl (C, alignment = 1, size = 1)
b_global_decl (Iaa, alignment = 4, size = 672)
b_global_decl (Raaa, alignment = 8, size = 576)
b_global_decl (R, alignment = 8, size = 8)
b_func_prologue (main)
b_push_const_int (10)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC0)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dot, void)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC1 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_nl, void)
b_push_const_int (100)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC2)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dot, void)
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( == , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC4)
b_jump ( destination = .LC2 )
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC3 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_nl, void)
b_push_ext_addr (I)
b_push_const_int (15)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (1)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC5)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dot, void)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC6 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_nl, void)
b_push_const_int (10)
b_convert (signed long int -> unsigned char)
b_convert (unsigned char -> signed long int)
b_duplicate (signed long int)
b_push_ext_addr (C)
b_push_const_int (1)
b_convert (signed long int -> unsigned char)
b_assign (unsigned char)
b_convert (unsigned char -> signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC7)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_dot, void)
b_duplicate (signed long int)
b_push_ext_addr (C)
b_inc_dec (unsigned char, PRE-INC)
b_jump ( destination = .LC8 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_nl, void)
b_push_const_int (9)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (0)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC9)
b_push_ext_addr (I)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 8, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_dispatch (  == ,signed long int, 6, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_dispatch (  == ,signed long int, 4, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_dispatch (  == ,signed long int, 2, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_dispatch (  == ,signed long int, 0, .LC13, pop on jump )
b_jump ( destination = .LC13 )
b_jump ( destination = .LC12 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_even, void)
b_jump ( destination = .LC11 )
b_dispatch (  == ,signed long int, 9, .LC20, pop on jump )
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 7, .LC20, pop on jump )
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 5, .LC20, pop on jump )
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 3, .LC20, pop on jump )
b_jump ( destination = .LC20 )
b_dispatch (  == ,signed long int, 1, .LC20, pop on jump )
b_jump ( destination = .LC20 )
b_jump ( destination = .LC19 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_odd, void)
b_jump ( destination = .LC11 )
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( == , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC26)
b_push_ext_addr (I)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_duplicate (signed long int)
b_push_ext_addr (J)
b_push_const_int (1)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC27)
b_push_ext_addr (J)
b_deref (signed long int)
b_push_const_int (7)
b_arith_rel_op ( == , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC29)
b_jump ( destination = .LC27 )
b_jump ( destination = .LC30 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_j, void)
b_duplicate (signed long int)
b_push_ext_addr (J)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC28 )
b_pop ()
b_push_ext_addr (I)
b_deref (signed long int)
b_push_const_int (99)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC31)
b_jump ( destination = .LC9 )
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC10 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_nl, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_i_j_values, void)
b_push_const_int (1)
b_duplicate (signed long int)
b_push_ext_addr (J)
b_push_const_int (10)
b_assign (signed long int)
b_arith_rel_op ( > , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC32)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_i_j_values, void)
b_duplicate (signed long int)
b_push_ext_addr (J)
b_inc_dec (signed long int, PRE-DEC)
b_jump ( destination = .LC33 )
b_pop ()
b_push_const_int (9)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (0)
b_assign (signed long int)
b_arith_rel_op ( < , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC34)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-INC)
b_jump ( destination = .LC35 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_push_const_int (0)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_push_const_int (10)
b_assign (signed long int)
b_arith_rel_op ( > , signed long int)
b_cond_jump (signed long int, NON-ZERO,
             .LC36)
b_push_ext_addr (I)
b_deref (signed long int)
b_dispatch (  == ,signed long int, 10, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 8, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 6, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 4, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 2, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_dispatch (  == ,signed long int, 0, .LC40, pop on jump )
b_jump ( destination = .LC40 )
b_jump ( destination = .LC39 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_even, void)
b_jump ( destination = .LC38 )
b_dispatch (  == ,signed long int, 9, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_dispatch (  == ,signed long int, 7, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_dispatch (  == ,signed long int, 3, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_dispatch (  == ,signed long int, 1, .LC48, pop on jump )
b_jump ( destination = .LC48 )
b_jump ( destination = .LC47 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_odd, void)
b_jump ( destination = .LC38 )
b_dispatch (  == ,signed long int, 5, .LC54, pop on jump )
b_jump ( destination = .LC54 )
b_jump ( destination = .LC53 )
b_jump ( destination = .LC36 )
b_jump ( destination = .LC38 )
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_duplicate (signed long int)
b_push_ext_addr (I)
b_inc_dec (signed long int, PRE-DEC)
b_jump ( destination = .LC37 )
b_pop ()
b_func_epilogue (main)
