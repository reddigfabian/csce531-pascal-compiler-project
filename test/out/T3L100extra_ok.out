..........
..........
...............
..........
EOjjEjjjOjjjjEjjjjjOjjjjjjEjjjjjjOjjjjjj
i = 10; j = 7
i = 10; j = 10
i = 10; j = 9
i = 10; j = 8
i = 10; j = 7
i = 10; j = 6
i = 10; j = 5
i = 10; j = 4
i = 10; j = 3
i = 10; j = 2
i = 10; j = 1
oops
Eoops
Ooops
Eoops
Ooops
Eoops
